homeassistant/components/nissan_leaf/binary_sensor.py

```diff
@@ -25,40 +29,40 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class LeafPluggedInSensor(LeafEntity, BinarySensorEntity):
     """Plugged In Sensor class."""
 
+    _attr_device_class = DEVICE_CLASS_PLUG
+
     @property
     def name(self):
         """Sensor name."""
         return f"{self.car.leaf.nickname} Plug Status"
 
+    @property
+    def available(self):
+        """Sensor availability."""
+        return self.car.data[DATA_PLUGGED_IN] is not None
+
     @property
     def is_on(self):
         """Return true if plugged in."""
         return self.car.data[DATA_PLUGGED_IN]
 
-    @property
-    def icon(self):
-        """Icon handling."""
-        if self.car.data[DATA_PLUGGED_IN]:
-            return "mdi:power-plug"
-        return "mdi:power-plug-off"
-
 
 class LeafChargingSensor(LeafEntity, BinarySensorEntity):
     """Charging Sensor class."""
 
+    _attr_device_class = DEVICE_CLASS_BATTERY_CHARGING
```

Comment:
Please separate new features from fixes in different PRs.

Reply:
Done.